src/main/java/com/iexec/blockchain/broker/BrokerService.java

```diff
@@ -126,9 +124,29 @@ public String matchOrders(BrokerOrder brokerOrder) {
 
     Optional<String> fireMatchOrders(BrokerOrder brokerOrder) {
         try {
-            FillOrdersCliOutput dealResponse = brokerClient.matchOrders(brokerOrder);
-            log.info("Matched orders [chainDealId:{}, tx:{}]", dealResponse.getDealid(), dealResponse.getTxHash());
-            return Optional.of(dealResponse.getDealid());
+            TransactionReceipt receipt = iexecHubService.
+                    getHubContract()
+                    .matchOrders(
+                            brokerOrder.getAppOrder().toHubContract(),
+                            brokerOrder.getDatasetOrder().toHubContract(),
+                            brokerOrder.getWorkerpoolOrder().toHubContract(),
+                            brokerOrder.getRequestOrder().toHubContract()
+                    ).send();
+            log.info("block {}, hash {}, status {}", receipt.getBlockNumber(), receipt.getTransactionHash(), receipt.getStatus());
+            log.info("logs count {}", receipt.getLogs().size());
+
+            List<String> events = IexecHubContract.getSchedulerNoticeEvents(receipt)
+                    .stream()
+                    .filter(event -> brokerOrder.getWorkerpoolOrder().getWorkerpool().equals(event.workerpool))
```

Comment:
What about the following to avoid any repetition of `brokerOrder.getWorkerpoolOrder().getWorkerpool()`?
```suggestion
            final String workerpool = brokerOrder.getWorkerpoolOrder().getWorkerpool();
            final List<String> events = IexecHubContract.getSchedulerNoticeEvents(receipt)
                    .stream()
                    .filter(event -> workerpool.equals(event.workerpool))
```

Reply:
I went for another evolution, please see 3a470e8

Reply:
2 points:
- Should we also use `requestOrder` at lines [98](https://github.com/iExecBlockchainComputing/iexec-blockchain-adapter-api/blob/3a470e8e2793f9a9d0b1df0614a3b78005cdd9e5/src/main/java/com/iexec/blockchain/broker/BrokerService.java#L98) & [115](https://github.com/iExecBlockchainComputing/iexec-blockchain-adapter-api/blob/3a470e8e2793f9a9d0b1df0614a3b78005cdd9e5/src/main/java/com/iexec/blockchain/broker/BrokerService.java#L115)?
- We still have a `workerpoolOrder#getWorkerpool` call in the `Stream#filter` pipeline. Should we store this value so that we call it a single time?